abci/example/kvstore/kvstore.go

```diff
@@ -454,5 +455,5 @@ func (app *Application) substPrepareTx(blockData [][]byte) []*types.TxRecord {
 		}
 	}
 
-	return trs
+	return append(trs, removed...)
```

Comment:
This is better—but we still have the problem that `trs` is preallocated to the full size of `blockData`, so any we don't fill (e.g., with the removed ones) will be nil in the original segment.